src/jquery.mockjax.js

```diff
@@ -269,6 +269,7 @@
 				url: mockHandler.proxy,
 				type: mockHandler.proxyType,
 				data: mockHandler.data,
+				async: false,
```

Comment:
Right, I see how that could cause problems. That said, we should not switch the `async` flag for _every_ proxy request. All I'm suggesting is to match that flag to what was in the original settings. Doing so should solve your problem above of having a synchronous ajax call in the source since the proxy would then also be synchronous.